frontend/catalyst/device/qjit_device.py

```diff
@@ -182,9 +182,9 @@ def extract_backend_info(
     if not pathlib.Path(device_lpath).is_file():
         raise CompileError(f"Device at {device_lpath} cannot be found!")
 
-    if hasattr(device, "shots"):
-        shots = get_device_shots(device) or 0
-        device_kwargs["shots"] = shots
+    # if hasattr(device, "shots"):
+    #    shots = get_device_shots(device) or 0
+    #    device_kwargs["shots"] = shots
```

Comment:
Since shots is no longer in the device attributes dictionary, [the lightning device shots parsers will all get zero](https://github.com/PennyLaneAI/pennylane-lightning/blob/master/pennylane_lightning/core/src/simulators/lightning_qubit/catalyst/LightningSimulator.hpp#L95).

To avoid issues, the following workflow should be adopted:
1. Open a PR in lightning to pin the catalyst git tag tracked by lightning to one before this PR gets merged.
2. Merge this PR. Lightning at this point still sees an old catalyst with the shots attr on the device init op.
3. Open a PR in lightning to remove shots attr parsing in the lightning devices. That PR tracks catalyst main back, aka remove the pin in step 1.

Reply:
I like the idea of having a friendly deprecation in between! Yes, my original proposal is a bit overly violent : )

We can think about the specifics later, but I like your outline.